src/term.c

```diff
@@ -1947,26 +1947,64 @@ static int cmd_sck(const PROGRAMMER *pgm, const AVRPART *p, int argc, char *argv
 
   if (argc == 1 && pgm->get_sck_period){
     if ((rc = pgm->get_sck_period(pgm, &v)) != 0) {
-      pmsg_error("(fosc) unable to get sck period (rc = %d)\n", rc);
+      pmsg_error("(fosc) unable to get SCK period (rc = %d)\n", rc);
       return -3;
     }
-    term_out("SCK period = %.1f us\n", v * 1e6 );
+    term_out("SCK period = %.1f us\n", v * 1e6);
+    term_out("SCK freq   = %d kHz\n", (int)(0.001/v));
     return 0;
   }
 
   if(argc != 2 || (argc > 1 && str_eq(argv[1], "-?"))) {
     msg_error(
       "Syntax: sck <value>\n"
-      "Function: set the SCK period\n"
+      "Function: set the SCK period in us or frequency in [kM]Hz\n"
     );
     return -1;
   }
+
   v = strtod(argv[1], &endp);
-  if (endp == argv[1]) {
-    pmsg_error("(sck) cannot parse period %s\n", argv[1]);
+  if (*endp != 0) {
```

Comment:
I think it is still important to check the case `endp == argv[1]` as this is indicates a missing number to start with.